Src/FluentAssertions/Execution/AssertionScope.cs

```diff
@@ -85,34 +85,42 @@ public AssertionScope(IAssertionStrategy assertionStrategy)
     /// and which will not throw until the scope is disposed.
     /// </summary>
     public AssertionScope(Lazy<string> context)
-        : this()
+        : this(new CollectingAssertionStrategy(), context)
     {
-        Context = context;
+        SetCurrentAssertionScope(this);
     }
 
     /// <summary>
     /// Starts a new scope based on the given assertion strategy and parent assertion scope
     /// </summary>
     /// <param name="assertionStrategy">The assertion strategy for this scope.</param>
-    /// <param name="parent">The parent assertion scope for this scope.</param>
     /// <exception cref="ArgumentNullException"><paramref name="assertionStrategy"/> is <see langword="null"/>.</exception>
-    private AssertionScope(IAssertionStrategy assertionStrategy, AssertionScope parent)
+    private AssertionScope(IAssertionStrategy assertionStrategy, Lazy<string> context)
     {
         this.assertionStrategy = assertionStrategy
             ?? throw new ArgumentNullException(nameof(assertionStrategy));
 
-        this.parent = parent;
+        parent = GetCurrentAssertionScope();
 
         if (parent is not null)
         {
             contextData.Add(parent.contextData);
-            Context = parent.Context;
             reason = parent.reason;
             callerIdentityProvider = parent.callerIdentityProvider;
             FormattingOptions = parent.FormattingOptions.Clone();
+            Context = new Lazy<string>(() => JoinContext(parent.Context, context));
+        }
+        else
+        {
+            Context = context;
         }
     }
 
+    private static string JoinContext(params Lazy<string>[] contexts)
```

Comment:
Suggestion on how to re-use context when possible.
```cs
private static Lazy<string> JoinContext(Lazy<string> parentContext, Lazy<string> context)
{
    if (parentContext is null)
    {
        return context;
    }

    if (context is null)
    {
        return parentContext;
    }

    return new(() => parentContext.Value + "/" + context.Value);
}
```

Reply:
Actually, they can both be `null`